### PR TITLE
Accessibility of generic segments tabs

### PR DESCRIPTION
**Pull request by mehreenmansur**
Issue link: [https://github.com/ComPlat/chemotion_ELN/issues/639](https://github.com/ComPlat/chemotion_ELN/issues/639) + https://github.com/ComPlat/chemotion_ELN/issues/795

https://user-images.githubusercontent.com/26220177/166630837-d7bb94b4-864d-488f-966d-95dab417d394.mp4

https://user-images.githubusercontent.com/26220177/166630840-ba3c9c87-084e-4e54-9f98-06adab0a255a.mp4

### CODE REVIEW

**Comment by nicolejung** [comment]
I have a few comments: 
1) could you add a hover over function the the red button that allow access to the collections and add here information on what you can so there? -> manage & organize collections: create or delete collections, adjust sharing options, adjust the visibility of tabs based on the collection level. something like this to explain the functionality

**Comment by nicolejung** [comment]
2. In the functionality itself: could you write a few sentences on what the user sees here and what can be done? e.g. opening the box with samples/reactions/screens/wellplate : you could add a section below explaining: "Here you can adjust how many and which segment tabs are visible in the selected collection....." Please explain a bit what happens after a change. 
3. Opening the drag&drop area: please explain that the user needs to drag into the white area to define the changes

**Comment by mehreenmansur** [comment]
segmentTab1
tab3

File: app/packs/src/components/collection_management/GenericSegmentsTabs.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+      }
+    });
+  }
+
+  onClickNode(node) {
+    this.setState({ currentCollection: node });
+    this.handleSelectModalOptions(this.state.selectModal);
+  }
+
+  getArrayFromLayout = (layout, availableTabs) => {
```
Comment (phuang26) [inline app/packs/src/components/collection_management/GenericSegmentsTabs.js:59]: duplicated code from ElementDetailSortTab, please dry the code

File: app/packs/src/components/fetchers/CollectionsFetcher.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import 'whatwg-fetch';
 import BaseFetcher from './BaseFetcher';
 import NotificationActions from '../actions/NotificationActions';
 import GenericElsFetcher from '../fetchers/GenericElsFetcher';
-import { downloadBlob } from '../utils/FetcherHelper';
+import { downloadBlob, getFileName } from '../utils/FetcherHelper';
```
Comment (phuang26) [inline app/packs/src/components/fetchers/CollectionsFetcher.js:5]: unused

File: app/packs/src/components/collection_management/GenericSegmentsTabs.js (hunk 1 above)
Comment (phuang26) [reply app/packs/src/components/collection_management/GenericSegmentsTabs.js:59]: also for other parts of this code

File: app/packs/src/components/collection_management/GenericSegmentsTabs.js [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+  }
+
+  handleSelectNav(eventKey) {
+    const currentTab = this.selectCurrentTab(eventKey);
+    let layout = {};
+    const node = this.state.currentCollection;
+    if (!isEmpty(node.tabs_segment[currentTab])){
+      layout = node.tabs_segment[currentTab];
+    } else {
+      layout = {
```
[the remark this answers is not in the document]
Comment (phuang26) [reply app/packs/src/components/collection_management/GenericSegmentsTabs.js:133]: if the layout comes from profile, then it includes all tabs already (including all segments)

File: app/packs/src/components/collection_management/GenericSegmentsTabs.js [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        analyses: -1,
+        literature: 3,
+        properties: 1,
+        qc_curation: 2,
+        references: -3,
+        results: -2
+      };
```
Comment (phuang26) [inline app/packs/src/components/collection_management/GenericSegmentsTabs.js:140]: the default layout should come from profile, not hard coding.
not all elements have the same tabs.
for example, reaction's tabs are different from samples.
if the layout comes from profile, then it includes all tabs already (including all segments)
screenshot-localhost_3000-2022 05 06-17_11_57
screenshot-localhost_3000-2022 05 06-17_12_13

**Review by phuang26 — COMMENTED** [review]
Steps: Collection Management > Collection Tabs > click on the collection > click on "Sample" --> nothing happen but got script error below:
```
ElementDetailSortTab.js:29 Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at getArrayFromLayout (ElementDetailSortTab.js:29:1)
    at CollectionTabs.handleSelectNav (CollectionTabs.js:103:1)
    at NavItem.handleClick (NavItem.js:46:1)
```

File: app/packs/src/components/CollectionTree.js [hunk 5]
```diff
@@ -37,7 +38,6 @@ export default class CollectionTree extends React.Component {
       syncCollectionVisible: false,
       inbox: inboxState.inbox,
       numberOfAttachments: inboxState.numberOfAttachments,
-      inboxVisible: false
```
Comment (phuang26) [inline app/packs/src/components/CollectionTree.js:40]: why remove this line?

Comment (phuang26) [inline app/packs/src/components/CollectionTree.js:40]: ?

**Review by phuang26 — APPROVED** [review]

**Review by FabianMauz — COMMENTED** [review]
WIP - more will come later

**Review by PiTrem — APPROVED** [review]

**Pull request merged**